File: internal/models/event.go
package models

import (
	"gorm.io/gorm"
	"math/rand"
	"time"
)

type Event struct {
    ID              uint           `gorm:"primaryKey" json:"id"`
    Title           string         `gorm:"not null;size:200" json:"title"`
    Description     string         `gorm:"type:text" json:"description"`
    Type            string         `gorm:"not null;size:50" json:"type"`
    DateTime        time.Time      `gorm:"not null" json:"date_time"`
    Location        string         `gorm:"not null" json:"location"`
    Latitude        float64        `json:"latitude"`
    Longitude       float64        `json:"longitude"`
    CreatorID       uint           `gorm:"not null" json:"creator_id"`
    Creator         User           `gorm:"foreignKey:CreatorID" json:"creator"`
    IsPrivate       bool           `gorm:"default:false" json:"is_private"`
    InviteCode  string `gorm:"size:20" json:"invite_code"`    // без uniqueIndex
    PrivateKey  string `gorm:"size:12" json:"private_key"`    // без uniqueIndex
    MaxParticipants int            `json:"max_participants"`
    CreatedAt       time.Time      `gorm:"autoCreateTime" json:"created_at"`
    UpdatedAt       time.Time      `gorm:"autoUpdateTime" json:"updated_at"`
    DeletedAt       gorm.DeletedAt `gorm:"index" json:"-"`

    IsSubscribed     bool  `gorm:"-" json:"is_subscribed"`
    SubscribersCount int64 `gorm:"-" json:"subscribers_count"`
    IsPast           bool  `gorm:"-" json:"is_past"`
}

type CreateEventRequest struct {
	Title           string `form:"title" binding:"required"`
	Description     string `form:"description" binding:"required"`
	Type            string `form:"type" binding:"required"`
	DateTime        string `form:"date_time" binding:"required"`
	Location        string `form:"location" binding:"required"`
	Latitude        string `form:"latitude"`
	Longitude       string `form:"longitude"`
	IsPrivate       bool   `form:"is_private"`
	MaxParticipants int    `form:"max_participants"`
}

// GenerateInviteCode генерирует уникальный код для приватного события
func (e *Event) GenerateInviteCode() {
	const charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
	code := make([]byte, 12)
	for i := range code {
		code[i] = charset[rand.Intn(len(charset))]
	}
	e.InviteCode = string(code)
}

// GeneratePrivateKey генерирует уникальный ключ для приватных событий
func (e *Event) GeneratePrivateKey() {
	const charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
	key := make([]byte, 8)
	for i := range key {
		key[i] = charset[rand.Intn(len(charset))]
	}
	e.PrivateKey = string(key)
}

File: SocialEvent/app/src/main/java/com/ark/socialevent/network/Event.kt
package com.ark.socialevent.network

import com.google.gson.annotations.SerializedName
import java.util.*

data class Event(
    val id: Int,
    val title: String,
    val description: String,
    val type: String,
    @SerializedName("date_time") val dateTime: String,
    val location: String,
    val latitude: Double?,
    val longitude: Double?,
    @SerializedName("creator_id") val creatorId: Int,
    val creator: UserProfile,
    @SerializedName("is_private") val isPrivate: Boolean,
    @SerializedName("invite_code") val inviteCode: String?,
    @SerializedName("private_key") val privateKey: String?,
    @SerializedName("max_participants") val maxParticipants: Int?,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String,
    @SerializedName("is_subscribed") val isSubscribed: Boolean = false,
    @SerializedName("subscribers_count") val subscribersCount: Int = 0,
    @SerializedName("is_past") val isPast: Boolean = false
)

data class CreateEventRequest(
    val title: String,
    val description: String,
    val type: String,
    @SerializedName("date_time") val dateTime: String,
    val location: String,
    val latitude: String? = null,
    val longitude: String? = null,
    @SerializedName("is_private") val isPrivate: Boolean = false,
    @SerializedName("max_participants") val maxParticipants: Int? = null
)

data class EventsResponse(
    val success: Boolean,
    val events: List<Event>? = null,
    val message: String? = null
)

data class EventResponse(
    val success: Boolean,
    val event: Event? = null,
    val message: String? = null
)

data class EventSubscriptionResponse(
    val success: Boolean,
    val message: String? = null
)